### PR TITLE
Upgrade to go 1.22.0

### PR DESCRIPTION
**Pull request by andrewpollock**

### CODE REVIEW

**Comment by andrewpollock** [comment]
@michaelkedar I'm going to have rebuild and release the CI image with [01d3a62](https://github.com/google/osv.dev/pull/1977/commits/01d3a6272eb15fd1aed9f3baef5be792dccf2cb0) in order to get this PR to pass, aren't I?

File: docker/indexer/go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/google/osv.dev/docker/indexer
 
-go 1.21.5
+go 1.22
```
Comment (michaelkedar) [inline docker/indexer/go.mod:3]: Do we want to keep the patch versions here?

**Review by michaelkedar — APPROVED** [review]
> @michaelkedar I'm going to have rebuild and release the CI image with [01d3a62](https://github.com/google/osv.dev/pull/1977/commits/01d3a6272eb15fd1aed9f3baef5be792dccf2cb0) in order to get this PR to pass, aren't I?

I think so? It's worth a shot anyway.

File: docker/indexer/go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (andrewpollock) [reply docker/indexer/go.mod]: Done in db4ca2b

**Review by cuixq — APPROVED** [review]

**Comment by andrewpollock** [comment]
> I think so? It's worth a shot anyway.

Yep, that did the trick

**Pull request merged**